pkg/sysextutils/sysext_utils.go

```diff
@@ -38,23 +41,281 @@ func getID(name string) string {
 	return fmt.Sprintf("%x", hasher.Sum(nil))
 }
 
+// May be naive. Forgive me :)
+
+func adjustSymlinks(rootfsDir string) error {
+	return filepath.Walk(rootfsDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.Mode()&os.ModeSymlink != 0 {
+			target, err := os.Readlink(path)
+			if err != nil {
+				logging.LogError("Failed to read symlink: %s, error: %v", path, err)
+				return err
+			}
+			if filepath.IsAbs(target) {
+				newTarget := filepath.Join(rootfsDir, strings.TrimPrefix(target, "/"))
+				if err := os.Remove(path); err != nil {
+					logging.LogError("Failed to remove old symlink: %s, error: %v", path, err)
+					return err
+				}
+				if err := os.Symlink(newTarget, path); err != nil {
+					logging.LogError("Failed to create new symlink: %s -> %s, error: %v", path, newTarget, err)
+					return err
+				}
+				logging.Log("Updated symlink: %s -> %s", path, newTarget)
+			} else {
+				relativeTarget := filepath.Join(filepath.Dir(path), target)
+				if _, err := os.Stat(relativeTarget); os.IsNotExist(err) {
+					newTarget := filepath.Join(rootfsDir, target)
+					if err := os.Remove(path); err != nil {
+						logging.LogError("Failed to remove old symlink: %s, error: %v", path, err)
+						return err
+					}
+					if err := os.Symlink(newTarget, path); err != nil {
+						logging.LogError("Failed to create new symlink: %s -> %s, error: %v", path, newTarget, err)
+						return err
+					}
+					logging.Log("Updated relative symlink: %s -> %s", path, newTarget)
+				}
+			}
+		}
+		return nil
+	})
+}
+
+// isStaticallyLinked determines if the specified binary is statically linked.
```

Comment:
This comment states the obvious, and can be improved: uses elf to check a binary's interp section to determine if it's statically linked.